From d5509b8a81adb63d69af159397d68f9722d25c20 Mon Sep 17 00:00:00 2001
From: Olivier Moysan <olivier.moysan@foss.st.com>
Date: Mon, 30 Oct 2023 10:21:38 +0100
Subject: [PATCH 0597/1141] iio: adc: stm32-mdf: fix uninitialized variable
 warning

Fixes following warning in Android build:
../drivers/iio/adc/stm32-mdf-adc.c:674:3: warning: variable 'logd' is uninitialized when used here [-Wuninitialized]
../drivers/iio/adc/stm32-mdf-adc.c:649:29: note: initialize the variable 'logd' to silence this warning

Signed-off-by: Olivier Moysan <olivier.moysan@foss.st.com>
Change-Id: I9522cc7ec5a88319077046a4a0f75d37eec14576
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/336775
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 drivers/iio/adc/stm32-mdf-adc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/iio/adc/stm32-mdf-adc.c b/drivers/iio/adc/stm32-mdf-adc.c
index b97e7405d09c..e5e26b611084 100644
--- a/drivers/iio/adc/stm32-mdf-adc.c
+++ b/drivers/iio/adc/stm32-mdf-adc.c
@@ -646,7 +646,7 @@ static int stm32_mdf_compute_scale(struct device *dev, unsigned int decim,
 {
 	unsigned long max = ARRAY_SIZE(stm32_mdf_log_table);
 	unsigned int prime_factors[16];
-	unsigned int num, div, logd;
+	unsigned int num, div, logd = 0;
 	int i, j, scale;
 
 	/* Decompose decimation ratio D, as prime number factors, to compute log10(D) */
-- 
2.39.2

